cdap-watchdog-api/src/main/java/io/cdap/cdap/api/metrics/MetricsWriter.java

```diff
@@ -37,7 +37,7 @@
    *
    * @param metricsContext metricsContext to be used for the MetricsWriter
    */
-  void initialize(MetricsContext metricsContext);
+  void initialize(MetricsContext metricsContext, MetricsWriterContext metricsWriterContext);
```

Comment:
Can we have the `MetricsWriterContext` contains the `MetricsContext getMetricsContext()` method so that we only need to pass in one context object?

Reply:
done.